src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String[] STATISTICS_FIELDS = {"supply", "buy", "result"};
+    private static final int SUPPLY_POS = 0;
+    private static final int BUY_POS = 1;
+    private static final int RESULT_POS = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(readFromFile(fromFileName), toFileName);
+    }
+
+    private int[] readFromFile(String fromFileName) {
+        int[] statisticsValues = new int[3];
+
+        try (FileReader reader = new FileReader(fromFileName);
+                BufferedReader bufferedReader = new BufferedReader(reader)) {
+            String readLine = bufferedReader.readLine();
+            String[] readValue;
+
+            while (readLine != null) {
+                readValue = readLine.split(",");
+                if (readValue[0].equals(STATISTICS_FIELDS[SUPPLY_POS])) {
+                    statisticsValues[0] += Integer.parseInt(readValue[1]);
+                } else {
+                    statisticsValues[1] += Integer.parseInt(readValue[1]);
+                }
+                readLine = bufferedReader.readLine();
```

Comment:
The read-from-file method should simply read the data from the file and return it. The logic of creating a report must be moved to a separate method.